locallib.php

```diff
@@ -346,24 +359,112 @@ function zoom_get_sessions_for_display($meetingid) {
     return $sessions;
 }
 
+/**
+ * Get the next occurrence of a meeting.
+ *
+ * @param stdClass $zoom
+ * @return int The timestamp of the next occurrence of a recurring meeting or
+ *             0 if this is a recurring meeting without fixed time or
+ *             the timestamp of the meeting start date if this isn't a recurring meeting.
+ */
+function zoom_get_next_occurrence($zoom) {
+    global $DB;
+
+    // Prepare an ad-hoc request cache as this function could be called multiple times throughout a request
+    // and we want to avoid to make duplicate DB calls.
+    $cacheoptions = array(
+            'simplekeys' => true,
+            'simpledata' => true
+    );
+    $cache = cache::make_from_params(cache_store::MODE_REQUEST, 'zoom', 'nextoccurrence', array(), $cacheoptions);
+
+    // If the next occurrence wasn't already cached, fill the cache.
+    if (($cachednextoccurrence = $cache->get($zoom->id)) === false) {
+        // If this isn't a recurring meeting.
+        if (!$zoom->recurring) {
+            // Use the meeting start time.
+            $cachednextoccurrence = $zoom->start_time;
+
+            // Or if this is a recurring meeting without fixed time.
+        } else if ($zoom->recurrence_type == ZOOM_RECURRINGTYPE_NOTIME){
+            // Use 0 as there isn't anything better to return.
+            $cachednextoccurrence = 0;
+
+            // Otherwise we have a recurring meeting with a recurrence schedule.
+        } else {
+            // Get the calendar event of the next occurrence.
+            $selectclause = "modulename = :modulename AND instance = :instance AND (timestart + timeduration) >= :now";
+            $selectparams = array('modulename' => 'zoom', 'instance' => $zoom->id, 'now' => time());
+            $nextoccurence = $DB->get_records_select('event', $selectclause, $selectparams, 'timestart ASC', 'timestart', 0, 1);
+
+            // If we haven't got a single event.
+            if (count($nextoccurence) != 1) {
+                // Use 0 as there isn't anything better to return.
+                $cachednextoccurrence = 0;
+            }
+
+            // Use the timestamp of the event.
+            $nextoccurenceobject = reset($nextoccurence);
+            $cachednextoccurrence = $nextoccurenceobject->timestart;
```

Comment:
Just to fix the typo, does not resolve the "else" conversation.
```suggestion
            $nextoccurrenceobject = reset($nextoccurrence);
            $cachednextoccurrence = $nextoccurrenceobject->timestart;
```

Reply:
updated the spelling here